contracts/SuiBridge.sol

```diff
@@ -36,18 +36,26 @@ contract SuiBridge is
 
     address[] public supportedTokens;
 
+    // So _weth9 should be 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 on mainnet?
     function initialize(address[] memory _supportedTokens, address _vault, address _weth9)
         external
         initializer
     {
         __Ownable_init();
         __ReentrancyGuard_init();
         __Pausable_init();
+
+        // need to init own chain id
+
         supportedTokens = _supportedTokens;
         vault = IBridgeVault(_vault);
         weth9 = IWETH9(_weth9);
     }
 
+    // can we have two different functions:
+    // 1. handleTokenTransfer(u8 mesasage_version, bytes memory payload) and
+    // 2. handleEmergencyOp(u8 mesasage_version, bytes memory payload) 
+    // the merit is no need for one more decoding in line 62
```

Comment:
No more decoding yes, but then we won't be able to introduce new messages without transferring bridge ownership to a new committee that can decode the new message type.